### PR TITLE
Add a convenience header file to inherit from face graphs

### PR DESCRIPTION
**Pull request by sloriot**
## Summary of Changes

Ship all needed overloads and specializations into a header file. Much like `interface_macros.h`, the user needs only to define 3 macros per inheriting structure.

## Release Management

* Affected package(s): BGL
* Feature/Small Feature (if any): [here](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/BGL_inheritance_helper) -- preapproved by Andreas Fabri 13:05, 20 August 2020 (CEST) 
* Link to compiled documentation (obligatory for small feature) [User Manual](https://cgal.geometryfactory.com/~sloriot/doc/BGL_inheritance/BGL/index.html#BGLInheriting) + [Reference doc](https://cgal.geometryfactory.com/~sloriot/doc/BGL_inheritance/BGL/graph__traits__inheritance__macros_8h.html)
* License and copyright ownership: GF

### CODE REVIEW

**Comment by sloriot** [comment]
It is somehow a bug-fix as some cleaning I've done since 5.0 showed some issues with dynamic property maps.

**Comment by afabri** [comment]
Shouldn't you add a test  like  `BGL/test/BGL/graph_concept_Derived.cpp` and the same for the properties?

**Comment by afabri** [comment]
We should write something like.      In order for a type `FG`  to be model of  the concept `FaceGraph` the specialization of `boost::graph_traits<FG>` must exist.    When you derive a class from `FG`  this graph traits class does not match.  Therefore we ....

**Comment by sloriot** [comment]
If possible, this PR should target 5.1 as issues with dynamic properties are being exposed by #4579, The aforementioned PR makes `copy_face_graph` use dynamic maps, and since this function is used in a lot of places, it increases the risk of an issue for users inheriting from a structure that is a face graph concept. On the other hand, the fix is small and can be easily imported as isolated in a new header.

File: BGL/doc/BGL/BGL.txt [hunk 1]
```diff
@@ -635,10 +635,10 @@ to parameterize a topological sphere by first virtually cutting it into a topolo
 disk.
 
 \subsection BGLInheriting Inheriting from a Model of a Face Graph Concept
-Several classes are provided as model of the face graph concepts by \cgal. In case, a user wants to inherit
-from one of those classes (or any such model in general), a header file consuming some user defined macros
-is provided to make the inheriting class a valid model of the same concepts. See `CGAL/boost/graph/graph_traits_inheritance_macros.h`
-for more details.
+In order for a type `FG` to be a model of one face graph concept, a specialization of
```
Comment (afabri) [inline BGL/doc/BGL/BGL.txt:638]: of one of the face graph concepts

File: BGL/doc/BGL/BGL.txt [hunk 2]
```diff
@@ -635,10 +635,10 @@ to parameterize a topological sphere by first virtually cutting it into a topolo
...
+`boost::graph_traits<FG>` must exist. When you derive a class from `FG` this graph traits class specialization does not match.
+For such cases, a header file consuming some user defined macros is provided to make the inheriting class a valid model
```
Comment (afabri) [inline BGL/doc/BGL/BGL.txt:640]: inheriting -> derived

File: BGL/doc/BGL/BGL.txt [hunk 3]
```diff
@@ -635,10 +635,10 @@ to parameterize a topological sphere by first virtually cutting it into a topolo
...
+of the same concepts. See `CGAL/boost/graph/graph_traits_inheritance_macros.h` for more details.
```
Comment (afabri) [inline BGL/doc/BGL/BGL.txt:641]: concepts -> concept

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-228.shtml

**Comment by lrineau** [comment]
Compilation error on Travis: https://travis-ci.com/github/CGAL/cgal/jobs/375678247#L7600

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-3.shtml

**Comment by lrineau** [comment]
I will merge this PR once the `CHANGES.md` is amended.

**Pull request merged**